packages/core/src/app/shipping/Shipping.tsx

```diff
@@ -345,6 +424,10 @@ export function mapToShippingProps({
     );
     const countriesWithAutocomplete = ['US', 'CA', 'AU', 'NZ'];
 
+    const card = getCart() || undefined;
+    const stripeUpe = getPaymentMethod('card', PaymentMethodId.StripeUPE);
+    const stripeUpeLinkEnable = stripeUpe && stripeUpe.initializationData.enableLink && card && card.currency.code === 'USD';
```

Comment:
you can break all those conditions in separate lines, if that works you may remove the complexity hack 
ie:
``` ts
    const linkEnabled = stripeUpe?.initializationData.enableLink || false;
    const stripeUpeSupportedCurrency = cart?.currency.code === 'USD' || false;
    const stripeUpeLinkEnable = linkEnabled && stripeUpeSupportedCurrency;
```